app/assets/javascripts/charts.js

```diff
@@ -259,7 +260,12 @@ function get_pie_chart(div, url) {
       $.getJSON(url, function(data) {
         var target = $("#"+div+"-body");
         target.empty();
+        var hostsCount = 0;
+        $.each(data.values,function() {
+          hostsCount += this.data;
+        });
         expanded_pie(target, data.values);
+        $('.modal-title').empty().append( __('Fact distribution chart') + ' - <b>' + data.name + ' </b><small> ('+ Jed.sprintf(n__("%s host", "%s hosts", hostsCount), hostsCount) +')</small>');
```

Comment:
It seems it is like that by design. You wish to see all of the fact values and their respective percents - so many labels, instead of one main label.